postgraphiql/src/components/PostGraphiQL.js

```diff
@@ -152,6 +158,32 @@ class PostGraphiQL extends React.PureComponent {
       };
     }
 
+    /**
+     * If we were configured with a graphiqlAuthorizationEventOrigin then we will set up
+     * a message event listener to allow windows with that origin to update our Authorization header.
+     */
+    if (POSTGRAPHILE_CONFIG.graphiqlAuthorizationEventOrigin) {
+      // We want to support a space separated list of origins but avoid a false positive if someone is tricky.
+      const paddedOriginsList = ` ${POSTGRAPHILE_CONFIG.graphiqlAuthorizationEventOrigin} `;
+      console.log('Setting up listener for messages from', paddedOriginsList);
+      window.addEventListener("message", (event) => {
+          if (!paddedOriginsList.includes(event.origin)) {
+            console.error('window.postMessage received from an unauthorized origin: ', event.origin);
+            return;
+          }
+
+          if (event.data && event.data.Authorization) {
+            const val = JSON.stringify({ Authorization: event.data.Authorization });
```

Comment:
Since headers are case insensitive, they're typically handed around in lower case (as in `req.headers` in Node.js). I recommend sticking to this lower-case convention, or using a case-insensitive accessor.

Reply:
Hmm.. Okay, I was going along with the casing that was used in the stringified `headersText` value.  Should that also be changed to lowercase?

Reply:
That's effectively user text, which should be treated case-insensitively. A user could enter `"AuThOrIzAtIoN": "Bearer ..."` and it should still work. The keys on the `event.data` payload should be lower case/camelCase; e.g.

```suggestion
          if (event.data && event.data.authorization) {
            const val = JSON.stringify({ Authorization: event.data.authorization });
```

But should we also merge in the other previous headers or overwrite them?